src/lizaalert/courses/mixins.py

```diff
@@ -17,3 +20,66 @@ class TimeStampedModel(models.Model):
 
     class Meta:
         abstract = True
+
+
+class HideOrderNumberMixin:
+    """
+    Убирает поле order_number при создании, оставляет при редактировании.
+
+    Логика создания нумерации поля order_number не очевидна, поэтому при
+    создании порядка данный механизм скрыт от администратора. При редактировании
+    объекта администратору показаывается нумерация и появляется возможность поменять
+    порядок.
+    """
+
+    field = "order_number"
+
+    def get_fields(self, request, obj=None):
+        fields = super().get_fields(request, obj)
+        if not obj and self.field:
+            return list(filter(lambda f: f != self.field, fields))
+        return fields
+
+
+def order_number_mixin(step, parent_field, name_of_instance):
+    """Order number mixin setter."""
+
+    class SaveOrderingMixin(models.Model):
+        """Ordering mixin."""
+
+        class Meta:
+            abstract = True
+
+        order_number = models.PositiveSmallIntegerField(
+            verbose_name=f"порядковый номер {name_of_instance}", validators=[MinValueValidator(1)], blank=True
+        )
+
+        @property
+        def order_queryset(self):
+            """Queryset for ordering."""
+            cls = type(self)
+            filter_params = {parent_field: getattr(self, parent_field)}
+            return cls.objects.filter(**filter_params)
+
+        @property
+        def parent_order(self):
+            # родительский объект, у которого есть order_number
+            """Chapter order for further ordering."""
+            try:
+                return getattr(getattr(self, parent_field), "order_number")
+            except AttributeError:
+                return None
```

Comment:
```suggestion
            if parent := getattr(self, parent_field):
                return getattr(parent, "order_number")
            return None
```

Reply:
AttributeError: 'Course' object has no attribute 'order_number'

Reply:
`return getattr(getattr(self, parent_field), "order_number", None)`
```suggestion
            try:
                return getattr(getattr(self, parent_field), "order_number")
            except AttributeError:
                return None
```

Вспомнил про более красивое решение, мы же можем сказать getattr вернуть что то, в случае отсутствия аттрибута